File: typescript/src/data_models/user_task/user_task.ts
import {IIdentity} from '@essential-projects/iam_contracts';

import {UserTaskConfig} from './user_task_config';

/**
 * Describes a user task that belongs to an active correlation.
 */
export class UserTask {
  /**
   * The id of the UserTask.
   */
  public id: string;
  /**
   * The instance id of the UserTask.
   */
  public flowNodeInstanceId?: string;
  /**
   * The name of the UserTask.
   */
  public name: string;
  /**
   * The id of the correlation that the UserTask belongs to.
   */
  public correlationId: string;
  /**
   * The id of the process model that the UserTask belongs to.
   */
  public processModelId: string;
  /**
   * The id of the process instance that the UserTask belongs to.
   */
  public processInstanceId?: string;
  /**
   * Contains information about the UserTask configuration, such as the form fields that can be addressed.
   */
  public data: UserTaskConfig;
  /**
   * The token payload the UserTask got suspended with.
   */
  public tokenPayload: any;
  /**
   * Contains the identity the UserTask belongs to.
   */
  public owner: IIdentity;
}

File: dotnet/src/data_models/user_task/UserTask.cs
namespace ProcessEngine.ConsumerAPI.Contracts.DataModel
{
    /// <summary>
    /// Describes a UserTask that belongs to an active Correlation.
    /// </summary>
    public class UserTask
    {
        /// <summary>
        /// The ID of the UserTask.
        /// </summary>
        public string Id { get; set; }

        /// <summary>
        /// The FlowNodeInstance ID of the UserTask.
        /// </summary>
        public string FlowNodeInstanceId { get; set; }

        /// <summary>
        /// The name of the UserTask.
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// The ID of the Correlation that the UserTask belongs to.
        /// </summary>
        public string CorrelationId { get; set; }

        /// <summary>
        /// The ID of the ProcessModel that the UserTask belongs to.
        /// </summary>
        public string ProcessModelId { get; set; }

        /// <summary>
        /// The ID of the ProcessInstance that the UserTask belongs to.
        /// </summary>
        public string ProcessInstanceId { get; set; }

        /// <summary>
        /// Contains information about the UserTask configuration, such as the form fields that can be addressed.
        /// </summary>
        public UserTaskConfig Data { get; set; }

        /// <summary>
        /// The token payload the UserTask got suspended with.
        /// </summary>
        public object TokenPayload { get; set; }
    }
}
